### ISSUE TITLE
Select word w/o copying it

### ISSUE THREAD

**Issue by wasker** [alefragnani/vscode-copy-word#11]
That'd be great to be able to just select the word, without copying it.

**Comment by alefragnani** [comment]
Hi @wasker ,

Do you know the native `Add Selection to Next Find Match` command does exactly this? 

Just hit `Cmd + D` (on Mac) / `Ctrl + D` (on Win/Linux) and it will select the word in the cursor position. BTW, if you hit the same command again and again, it will select the other occurrences of the same word.

Hope this helps

**Comment by wasker** [comment]
I would never know by the name of it... Thanks for pointing me there!

**Comment by alefragnani** [comment]
Me neither, but this keyboard shortcut was one of the first _productivity features_ I’ve learned in VSCode, so I use it a lot 😁 .

**Comment by wasker** [comment]
I wonder if you'd reconsider reopening this issue. Ctrl-D is better than nothing, but it doesn't behave like Ctrl-Shift-W in Visual Studio 2019. :( It doesn't really select *words*, and it looks like behavior in general depends on the language in the editor. E.g., if I Ctrl-D on "workbench" in the file below, the whole string with double-quotes gets selected.

```
{ "key": "ctrl+shift+s",       "command": "workbench.action.files.saveAll" }
```

**Comment by wasker** [comment]
Also, sorry to bug *you* - it's just you're the person who already has extension for operations on words, and simple word selection seem to be logical to have there too. :)

**Comment by alefragnani** [comment]
No problem, you are not _bugging me_ 😁 .

I'm not familiar with VS2019, so I don't know how it works. I made a search and found [this documentation](https://docs.microsoft.com/pt-br/visualstudio/ide/default-keyboard-shortcuts-in-visual-studio?view=vs-2019), which says different things about the commands. Maybe you used a different release or has some plugin installed.

| Command | Shortcut |
|------------|---------|
| Edit.SelectCurrentWord | `Ctrl + W`|
| File.ViewInBrowser | `Ctrl + Shift + W` |

Also, my extension has the same _selection behaviour_ as `Ctrl + D`, while copying/cutting the example you provided, so it will not work for you either. 

What you need, in fact, is a new Select Word algorithm (probably extracted from VS2019), which appears to recognise only letters, numbers and underscore as _valid word characters_. Maybe a few other symbols/combinations...

Hope this helps

**Comment by alefragnani** [comment]
It appears your request has also been made in VS Code itself..

* https://github.com/microsoft/vscode/issues/87310
* https://github.com/microsoft/vscode/issues/15774

**Comment by wasker** [comment]
Thanks for the pointers!

I was referring to Edit.SelectCurrentWord behavior. It does look like VSCode needs to have separate functionality. I'll file an issue there.

**Comment by wasker** [comment]
Available for all in 1.44.
